docs/readconsoleinputex.md

```diff
@@ -0,0 +1,161 @@
+- ReadConsoleInputExW
+api_location:
+- Kernel32.dll
+- API-MS-Win-Core-Console-l1-1-0.dll
+- KernelBase.dll
+- API-MS-Win-DownLevel-Kernel32-l1-1-0.dll
+- MinKernelBase.dll
+api_type:
+- DllExport
+---
+
+# ReadConsoleInputEx function
+
+Reads data from a console input buffer and removes it from the buffer, with configurable behavior.
+
+<div class="alert"><b>Note</b> This function has no associated import library. This function is available as the resources named <b>ReadConsoleInputExA</b> and <b>ReadConsoleInputExW</b> in Kernel32.dll. You must use the <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-loadlibrarya">LoadLibrary</a> and <a href="/windows/desktop/api/libloaderapi/nf-libloaderapi-getprocaddress">GetProcAddress</a> functions to dynamically link to Kernel32.dll.</div><div> </div>
+
+## Syntax
+
+```C
+BOOL WINAPI ReadConsoleInputEx(
+  _In_  HANDLE        hConsoleInput,
+  _Out_ PINPUT_RECORD lpBuffer,
+  _In_  DWORD         nLength,
+  _Out_ LPDWORD       lpNumberOfEventsRead,
+  _In_  USHORT        wFlags
+);
+```
+
+## Parameters
+
+*hConsoleInput* \[in\]  
+A handle to the console input buffer. The handle must have the **GENERIC\_READ** access right. For more information, see [Console Buffer Security and Access Rights](console-buffer-security-and-access-rights.md).
+
+*lpBuffer* \[out\]  
+A pointer to an array of [**INPUT\_RECORD**](input-record-str.md) structures that receives the input buffer data.
+
+*nLength* \[in\]  
+The size of the array pointed to by the *lpBuffer* parameter, in array elements.
+
+*lpNumberOfEventsRead* \[out\]  
+A pointer to a variable that receives the number of input records read.
+
+*wFlags* \[in\]
+
+A set of flags (ORed together) that specify the console's reading behavior.
+
+| Value | Meaning |
+|-|-|
+| **CONSOLE_READ_NOREMOVE** `0x0001` | Leave the events in the input buffer (as in [`PeekConsoleInput`](peekconsoleinput.md)) |
+| **CONSOLE_READ_NOWAIT** `0x0002` | Return immediately, even if there are no events in the input buffer.
+
+## Return value
+
+If the function succeeds, the return value is nonzero.
+
+If the function fails, the return value is zero. To get extended error information, call [**GetLastError**](/windows/win32/api/errhandlingapi/nf-errhandlingapi-getlasterror).
+
+## Remarks
+
+This function is a configurable version of `ReadConsoleInput`. See the [remarks for `ReadConsoleInput`](readconsoleinput.md) for additional operational details.
+
+Calling `ReadConsoleInputEx` with the flags `CONSOLE_READ_NOMOVE | CONSOLE_READ_NOWAIT` is equivalent to calling [`PeekConsoleInput`](peekconsoleinput.md).
+
+This function does not exist in the Windows Console headers. To gain access to it from a C or C++ application, include the following declarations and dynamically link kernel32.dll as noted above.
+
+```c
+#ifndef CONSOLE_READ_NOREMOVE
+#define CONSOLE_READ_NOREMOVE   0x0001
+#endif
+
+#ifndef CONSOLE_READ_NOWAIT
+#define CONSOLE_READ_NOWAIT     0x0002
+#endif
+
```

Comment:
Technically you should probably put in a...

```
#ifdef UNICODE
ReadConsoleInputEx ReadConsoleInputExW
#else
ReadConsoleInputEx ReadConsoleInputExA
#endif
```

... here too because that's what we would do if we were to put it into the headers.

Reply:
That's a fair call. In this case, it's not actually linkable via those names but... perhaps it's worth noting anyway

Reply:
Yeah maybe. Or I'm crazy. IDK. Just a suggestion